src/pages/_middleware.ts

```diff
@@ -20,13 +34,40 @@ const PUBLIC_FILE = /\.(.*)$/
  * https://github.com/vercel/examples/edge-functions/ab-testing-simple
  *
  */
-export function middleware(req: NextRequest) {
-  // Only rewrite files that don't have a file extension (think favicon)
+export async function middleware(req: NextRequest) {
+  let response = NextResponse.next()
+  // think favicon etc
+  if (PUBLIC_FILE.test(req.nextUrl.pathname)) return response
+
   // Only rewrite if we are at the root
-  if (req.nextUrl.pathname === '/' && !PUBLIC_FILE.test(req.nextUrl.pathname)) {
+  if (req.nextUrl.pathname === '/') {
+    const cioId =
+      req.cookies[CIO_COOKIE_KEY] ||
+      req.nextUrl.searchParams.get(CIO_COOKIE_KEY)
+
     // if there's a cookie with a token they are logged in
-    let status = req.cookies[ACCESS_TOKEN_KEY] ? 'logged_in' : 'anon'
-    req.nextUrl.pathname = status === 'logged_in' ? `/` : `/signup`
-    return NextResponse.rewrite(req.nextUrl)
+    let status = cioId || req.cookies[ACCESS_TOKEN_KEY] ? 'identified' : 'anon'
+
+    switch (status) {
+      case 'anon':
+        response = NextResponse.rewrite('/signup')
+        break
+      case 'identified':
+        if (cioId) {
```

Comment:
an identified user can be logged in or out, but if we have CIO data we want to use it